### PR TITLE
Add Model Tensor Planning (MTP) algorithm

### PR DESCRIPTION
**Pull request by anindex**
This PR adds **Model Tensor Planning** (MTP) as a new sampling-based controller, together with a U-maze navigation task that showcases its global exploration capability.

**Paper:** [Model Tensor Planning](https://arxiv.org/abs/2505.01059) - Le, Nguyen, Vu, Carvalho, Peters, TMLR 2025

### What is MTP?

MTP injects structured global exploration into a CEM-style local update. At each planning step it:

1. Samples `β · num_samples` paths through an M-partite tensor graph, each path is M independent waypoints, smoothed by Akima, B-spline, or linear interpolation into a control trajectory.
2. Samples the remaining `(1 − β) · num_samples` from a local Gaussian around the current best plan.
3. Selects elites via `jax.lax.top_k` and updates the mean/variance with softmax-weighted, Bessel-corrected statistics.

This lets MTP escape local minima that defeat PS / MPPI / CEM (e.g., a wall between start and goal).

### Files

| File | Description |
|------|-------------|
| `hydrax/algs/mtp.py` | MTP controller - `MTP` class and `MTPParams` dataclass |
| `hydrax/utils/spline.py` | Akima cubic + B-spline basis matrix primitives (appended to existing module) |
| `hydrax/tasks/navigation.py` | U-maze navigation task (point mass with inner walls) |
| `hydrax/models/particle_navigation/` | MJCF assets for the navigation task |
| `examples/navigation.py` | Interactive demo - supports `ps`, `mppi`, `cem`, `mtp` subcommands |
| `tests/test_mtp.py` | Algorithm tests: open-loop optimisation, shape checks, edge cases |
| `tests/test_mtp_splines.py` | Spline tests: knot interpolation, Akima vs linear MSE, B-spline partition of unity |
| `tests/test_navigation.py` | Task smoke test (both `jax` and `warp` backends) |

### Design decisions

- **Spline functions live in `hydrax/utils/spline.py`** rather than a separate subpackage - keeps the module hierarchy flat and lets any future algorithm reuse them.
- **Tensor sampling is simplified:** instead of materialising an `(M, N, nu)` graph and gathering paths via random indices, we sample `(mtp_samples, M, nu)` directly from uniform. Mathematically identical when candidates per layer are i.i.d., saves one allocation + gather.
- **`nan_to_num` removed from softmax weights:** `jax.nn.softmax` is already logsumexp-stable; silencing NaN would hide upstream cost bugs. Consistent with every other Hydrax algorithm.
- **`best_knots` is warm-started** in an `optimize()` override - the base class only re-interpolates `mean` onto the new knot time grid; we also re-interpolate `best_knots` to prevent control discontinuities when `spline_type != "zero"`.
- **Parameter naming** uses `m_pts` / `n_per_layer` (PEP 8) instead of the paper's uppercase `M` / `N`.

### Usage

```python
from hydrax.algs import MTP
from hydrax.tasks.navigation import Navigation

task = Navigation()
ctrl = MTP(
    task,
    num_samples=64,
    m_pts=3,
    n_per_layer=50,
    beta=0.5,
    mtp_interpolation="akima",
    plan_horizon=0.5,
    spline_type="zero",
    num_knots=11,
)

### CODE REVIEW

**Comment by vincekurtz** [comment]
Hi An, thanks for this contribution! Could you split this into three separate PRs?

1. New spline utils
2. MTP solver itself
3. The navigation/bug trap example

We can land these in order, and rebase after each one.

**Comment by anindex** [comment]
Close for incremental PRs.
